components/action-mgt/org.wso2.carbon.identity.action.management/src/main/java/org/wso2/carbon/identity/action/management/util/ActionMgtV2AuditLoggerUtil.java

```diff
@@ -0,0 +1,248 @@
+import static org.wso2.carbon.identity.central.log.mgt.utils.LogConstants.ActionManagement.UPDATE_ACTION;
+import static org.wso2.carbon.identity.central.log.mgt.utils.LogConstants.ActionManagement.USERNAME_FIELD;
+import static org.wso2.carbon.identity.central.log.mgt.utils.LoggerUtils.jsonObjectToMap;
+import static org.wso2.carbon.identity.central.log.mgt.utils.LoggerUtils.triggerAuditLogEvent;
+
+/**
+ * Action management V2 audit logger util class.
+ */
+public class ActionMgtV2AuditLoggerUtil {
+
+    /**
+     * To get the current user, who is doing the current task.
+     *
+     * @return current logged-in user.
+     */
+    private static String getUser() {
+
+        String user = CarbonContext.getThreadLocalCarbonContext().getUsername();
+        if (StringUtils.isNotEmpty(user)) {
+            user = UserCoreUtil
+                    .addTenantDomainToEntry(user, CarbonContext.getThreadLocalCarbonContext().getTenantDomain());
+        } else {
+            user = CarbonConstants.REGISTRY_SYSTEM_USERNAME;
+        }
+        return user;
+    }
+
+    /**
+     * Get the initiator for audit logs.
+     *
+     * @return Initiator id despite masking.
+     */
+    private static String getInitiatorId() {
+
+        String initiator = null;
+        String username = MultitenantUtils.getTenantAwareUsername(getUser());
+        String tenantDomain = MultitenantUtils.getTenantDomain(getUser());
+        if (StringUtils.isNotBlank(username) && StringUtils.isNotBlank(tenantDomain)) {
+            initiator = IdentityUtil.getInitiatorId(username, tenantDomain);
+        }
+        if (StringUtils.isBlank(initiator)) {
+            if (username.equals(CarbonConstants.REGISTRY_SYSTEM_USERNAME)) {
+                // If the initiator is wso2.system, we need not to mask the username.
+                return LoggerUtils.Initiator.System.name();
+            }
+            initiator = LoggerUtils.getMaskedContent(getUser());
+        }
+        return initiator;
+    }
+
+    /**
+     * Generate the audit log entry related to the operation.
+     *
+     * @param operation  Operation to be logged.
+     * @param actionType Type of action.
+     * @param actionId   ID of action.
+     * @param action     Action to be logged.
+     */
+    public JSONObject generateAuditLogEntry(String operation, String actionType, String actionId, Action action) {
+
+        JSONObject data = null;
+        switch (operation) {
+            case ADD_ACTION:
+                data = createAuditLogsForAddAction(action);
+                break;
+            case UPDATE_ACTION:
+                data = createAuditLogsForUpdateAction(actionId, action);
+                break;
+            case DELETE_ACTION:
+            case ACTIVATE_ACTION:
+            case DEACTIVATE_ACTION:
+                data = generateAuditLogsById(actionType, actionId);
+                break;
+        }
+        return data;
+    }
```

Comment:
Do we really need to make this public. We can still use this from a single log method IMO.